src/App.js

```diff
@@ -1,19 +1,60 @@
-import React, { useState } from 'react'
+import React, { useEffect, useState } from 'react'
+import { Route, Switch } from 'react-router-dom'
+import { v4 as uuidv4 } from 'uuid'
+import AddNewProjectTab from './components/AddNewProjectTab'
+import LogoHeader from './components/LogoHeader'
 import ProjectList from './components/ProjectList'
+import ShoppingList from './components/ShoppingList'
 import WelcomeScreen from './components/WelcomeScreen.js'
 
 export default function App() {
-  const [showProjectTab, setShowProjectTab] = useState(false)
+  const [projectList, setProjectList] = useState([])
+
+  useEffect(() => {
+    setProjectList(JSON.parse(localStorage.getItem('projects') || '[]'))
+  }, [])
+
+  useEffect(() => {
+    localStorage.setItem('projects', JSON.stringify(projectList))
+  }, [projectList])
 
   return (
-    <>
-      {showProjectTab ? (
-        <>
-          <ProjectList />
-        </>
-      ) : (
-        <WelcomeScreen startApp={() => setShowProjectTab(true)} />
-      )}
-    </>
+    <Switch>
+      <Route exact path="/">
+        <WelcomeScreen />
+      </Route>
+      <Route path="/shopping-list">
+        <LogoHeader />
+        <ShoppingList projectList={projectList} />
```

Comment:
Du könntest Dir überlegen, ob Du die Komponenten in Pages baust, wenn Du magst. Das macht es vielleicht noch einen Tick übersichtlicher :)